src/lib/decorators.ts

```diff
@@ -225,6 +225,54 @@ export function query(selector: string) {
   };
 }
 
+/**
+ * A property decorator that converts a class property into a getter that
+ * returns a promise that resolves to the result of a querySelector on the
+ * element's renderRoot done after the element's `updateComplete` promise
+ * resolves and after the queried element's `updateComplete` promise resolves
+ * (if it is a LitElement).
+ *
+ * @param selector A DOMString containing one or more selectors to match.
+ *
+ * See: https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector
+ *
+ * @example
+ *
+ *     class MyElement {
+ *       @asyncQuery('#first')
+ *       first;
```

Comment:
I think the example should show the usage of this.first somewhere too.

Reply:
Done.